packages/icon/src/template.ts

```diff
@@ -0,0 +1,143 @@
+/**
+usage:
+cd packages/icon
+
+# for entire folder
+yarn svgr --template src/template.js -d dist/test src/glyphs
+
+# for one file while developing
+yarn svgr --template src/template.js -d dist/test src/glyphs/ActivityFeed.svg
+
+I have these tabs open atm:
+https://react-svgr.com/docs/custom-templates/
+https://github.com/gregberge/svgr/blob/master/packages/babel-plugin-transform-svg-component/src/index.js#L28
+https://babeljs.io/docs/en/babel-template#template-literal-usage
+https://astexplorer.net/
+*/
+
+/**
+ function createGlyphComponent(glyphName, Glyph) {
+  function GlyphComponent({ className, size = 16, title, fill, ...props }) {
+
+
+    return (
+      <Glyph
+        className={cx(
+          {
+            [fillStyle]: fill != null,
+          },
+          className,
+        )}
+        title={getGlyphTitle(glyphName, title)}
+        height={size}
+        width={size}
+        {...props}
+      />
+    );
+  }
+}
+**/
+interface FirstArg extends Record<string, any> {
+  template: {
+    smart: (opts: Record<string, any>) => Record<string, any>;
+  } & Record<string, any>;
+}
+
+interface SecondArg extends Record<string, any> {
+  state: {
+    componentName: string;
+  } & Record<string, any>;
+  typescript: boolean;
+}
+
+interface ThirdArg extends Record<string, any> {
+  jsx: Record<string, any>;
+  componentName: object;
+  imports: string;
+  exports: string;
+}
+
+module.exports = function template(
+  { template }: FirstArg,
+  { state: { componentName }, typescript }: SecondArg,
+  { imports, jsx, exports }: ThirdArg,
+) {
+  const plugins = ['jsx'];
+
+  if (typescript) {
+    plugins.push('typescript');
+  }
+
+  const typeScriptTpl = template.smart({ plugins });
+
+  const jsxAttributes = typeScriptTpl.ast`
+    <Glyph
+      className={cx(
+        {
+          [fillStyle]: fill != null,
+        },
+        className,
+      )}
+      title={getGlyphTitle('${componentName}', title)}
+      height={size}
+      width={size}
+    />`;
+
+  // Augment the `<svg attributes />` so we can customize it with the values above.
+  jsx.openingElement.attributes = jsxAttributes.expression.openingElement.attributes.concat(
+    jsx.openingElement.attributes[2],
+  );
+
+  return typeScriptTpl.ast`
+    ${imports}
+    import PropTypes from 'prop-types';
+    import { css, cx } from '@leafygreen-ui/emotion';
+
+    function getGlyphTitle(name, title) {
```

Comment:
A size perf idea I had was to eventually move this function into its own file and import it, instead of re-creating the function every time.

Can do it later though.

Reply:
I think this is actually pretty easy to do now, just need to make a new file that exports this function and update this template to import it. TypeScript will transpile that file for us automatically now so it'll be a nice and easy relative import.